src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/service/ResyncAlertsService.kt

```diff
@@ -0,0 +1,91 @@
+package uk.gov.justice.digital.hmpps.hmppsalertsapi.service
+
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.entity.Alert
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.entity.AlertCode
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.enumeration.Source
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.model.ResynchedAlert
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.model.request.ResyncAlert
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.repository.AlertCodeRepository
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.repository.AlertRepository
+import java.util.UUID
+
+@Transactional
+@Service
+class ResyncAlertsService(
+  private val alertRepository: AlertRepository,
+  private val alertCodeRepository: AlertCodeRepository,
+) {
+  fun resyncAlerts(prisonNumber: String, alerts: List<ResyncAlert>): List<ResynchedAlert> {
+    val existingAlerts = alertRepository.findByPrisonNumber(prisonNumber)
+    existingAlerts.forEach {
+      it.delete(
+        deletedBy = "SYS",
+        deletedByDisplayName = "SYS",
+        source = Source.NOMIS,
+        activeCaseLoadId = null,
+        description = "Alert deleted via resync",
+      )
+      alertRepository.saveAll(existingAlerts)
+    }
+    val alertCodes = getValidatedAlertCodes(alerts)
```

Comment:
As this is transactional, it'll roll back in the event that an alert code is invalid however it's easier to read if the validation checks are first